util/helm/cmd.go

```diff
@@ -63,6 +63,8 @@ func (c Cmd) run(args ...string) (string, error) {
 
 	if c.IsHelmOci {
 		cmd.Env = append(cmd.Env, "HELM_EXPERIMENTAL_OCI=1")
+		// set the home for registry login/logout to use it as the registry/config.json location
+		cmd.Env = append(cmd.Env, fmt.Sprintf("HELM_CONFIG_HOME=%s/config", c.helmHome))
```

Comment:
Looks like this broke when upgrading to Helm v3. If you see above we had set `HELM_HOME`. But in v3 they moved to `HELM_CONFIG_HOME`:

https://helm.sh/docs/topics/v2_v3_migration/#overview-of-helm-3-changes
> v3 configuration no longer uses $HELM_HOME and uses XDG directory specification instead. It is also created on the fly as need be. It is therefore independent of v2 configuration. This is applicable only when both versions are installed on the same system

So I think the proper fix is to replace `HELM_HOME` with `HELM_CONFIG_HOME` above so it works for OCI and non-OCI

Reply:
> and uses XDG directory specification instead

It may have been worked for non-OCI because perhaps non-OIC implementation understand XDG directory specification whereas OCI implementation may have not. Regardless, setting `HELM_HOME` has no effect and setting `HELM_CONFIG_HOME` in the common codepath seems like the right thing to do.